### DISCUSSION TITLE
Running LDT using SMAP_L3 3 KM

### DISCUSSION THREAD

**Discussion by MariRiba** [NASA-LIS/LISF#1454] [Q&A]
Hello ,
I am trying to run LDT using SMAP_L3 3 KM datasets [ from this website](https://n5eil01u.ecs.nsidc.org/SMAP/SPL3SMA.003/) Does the latest version ( LISF-7.4.3-public ) support this ? If , no , then can I make changes in the codes to support this ?

I have tried to change */ldt/DAobs/NASA_SMAPsm/NASASMAPsm_obsMod.F90  , line 154

from 

        ` elseif(NASASMAPsmobs(n)%data_designation.eq."SPL3SMP_E") then
          **NASASMAPsmobs(n)%nc = 3856
          NASASMAPsmobs(n)%nr = 1624**
          gridDesci = 0
          gridDesci(1) = 9
          **gridDesci(2) = 3856
          gridDesci(3) = 1624**
          gridDesci(9) = 5 !M09 grid
          gridDesci(20) = 64
          **gridDesci(10) = 0.09**
          gridDesci(11) = 1 !for the global switch`
to

        ` elseif(NASASMAPsmobs(n)%data_designation.eq."SPL3SMP_E") then
          **NASASMAPsmobs(n)%nc = 11568
          NASASMAPsmobs(n)%nr = 4872**
          gridDesci = 0
          gridDesci(1) = 9
          **gridDesci(2) = 11568
          gridDesci(3) = 4872**
          gridDesci(9) = 5 !M09 grid
          gridDesci(20) = 64
          **gridDesci(10) = 0.03**
          gridDesci(11) = 1 !for the global switch`

With this changes I configured and compiled the LDT executable and it compiled successfully.

But I am getting this error when I am running the 5th step i.e 'DA_proc_SMAP'

 `[INFO] LDT cycle time: 04/16/2015 00:00:00`
  `[INFO] reading /scratch2/mari/lisf/SMAP3KM/RS_DATA/SMAP/SPL3SMP_E.003/2015.04.16 /SMAP_L3_SM_P_E_20150416_R18290_001.h5`
 `[ERR] Error opening SM group (AM) in NASASMAP file Stopping.  `        -1
 `endrun is being called` 

Please provide inputs on how to proceed with running the LDT step as mentioned above.
As there is nothing mentioned in the [user guide](https://nasa-lis.github.io/LISF/LDT_users_guide/LDT_users_guide.html)  regarding the same

**Comment by dmocko** [comment]
Hi @MariRiba 

No - LISF (ldt,lis,lvt) does not currently support the 3-km SMAP_L3 product.

It looks like you need to modify more code.  For example, LDT is trying read files that start with `SMAP_L3_SM_P_E`,
while the files for this product start with `SMAP_L3_SM_A`.

Modify the filename code (around lis_files) in this routine in the same directory: `readNASASMAPsmObs.F90`

I am also not sure why you have a space in your INFO statement right after the date, but before the "/".

**Reply by MariRiba** [reply]
Hi @dmocko 

Actually there is no space in between the INFO statement right after the date, but before the "/" , it was a typing mistake.

And I think that lots of changes should be made in the filename codes and function it is calling to be able to run the 3-km SMAP_L3 product. This is beyond my area of expertise.

Anyway , thanks for the help. I got lots of ideas trying to modify the codes .

**Reply by MariRiba** [reply]
Dear @dmocko ,

 We did the necessary changes in the code as suggested by you. Accordingly, we also made changes in the LIS files. We have basically tried to add a separate subroutine for 3 km SMAP data with the product name starting with   **SMAP_L3_SM_A.**

`** Question 1** `

Below I have attached the changes, and have also mentioned **`the file location`** and the **_`_original line number_`_** where I have made the changes. However, we are still getting the same error. The error message is attached at the end. What can be the reason for this? We have also attached the logfile.

**1. LISF/ldt/DAobs/NASA_SMAPsm/NASASMAPsm_obsMod.F90**

**1.i. Line 164**

![Screenshot from 2024-01-23 14-03-45](https://github.com/NASA-LIS/LISF/assets/138671730/1f5ca6b7-8900-4303-be85-234111e97953)


**2. LISF/ldt/DAobs/NASA_SMAPsm/readNASASMAPsmObs.F90**

**2.i. Line 176**
![Screenshot from 2024-01-23 14-07-24](https://github.com/NASA-LIS/LISF/assets/138671730/1842e2c2-6d81-40db-9ea8-3662cee5db8d)

![Screenshot from 2024-01-23 14-07-59](https://github.com/NASA-LIS/LISF/assets/138671730/a6cc23d1-3f9d-4be4-998c-2b5e3bf18b15)

**2.ii. Line 761**
![Screenshot from 2024-01-23 14-09-05](https://github.com/NASA-LIS/LISF/assets/138671730/a7885852-99a6-42e3-a2e0-95cb82c3a6de)

**3. LISF/lis/dataassim/obs/NASA_SMAPsm/NASASMAPsm_Mod.F90**

**3.i.Line 371**
![Screenshot from 2024-01-23 14-11-37](https://github.com/NASA-LIS/LISF/assets/138671730/ce892548-34d6-4af0-a61f-ba285e9a7e2f)


**3.ii.Line 589**

![Screenshot from 2024-01-23 14-12-01](https://github.com/NASA-LIS/LISF/assets/138671730/939c81c7-bd1c-4396-8f2d-1ce6bbfeb7de)

**3.iii.Line 615**

![Screenshot from 2024-01-23 14-12-41](https://github.com/NASA-LIS/LISF/assets/138671730/ec9e3802-1973-495c-9591-727a967b5670)


**4. LISF/lis/dataassim/obs/NASA_SMAPsm/read_NASASMAPsm.F90**

**4.i.Line 208**

![Screenshot from 2024-01-23 14-14-13](https://github.com/NASA-LIS/LISF/assets/138671730/beff12ff-1510-4f0a-aec3-900a8871657b)


![Screenshot from 2024-01-23 14-14-50](https://github.com/NASA-LIS/LISF/assets/138671730/cd77923e-22a2-4643-9a47-3efc4629a5b4)

**4.ii.Line 1519**

![Screenshot from 2024-01-23 14-16-49](https://github.com/NASA-LIS/LISF/assets/138671730/868c1a09-1358-4fbb-b6ca-7163730fc4f4)


Error message:- 
![Screenshot from 2024-01-23 15-22-07](https://github.com/NASA-LIS/LISF/assets/138671730/3702588b-9fac-4213-a416-71d898ba13c1)

Log file ;
[logfile.zip](https://github.com/NASA-LIS/LISF/files/14021807/logfile.zip)



**`Question 2`**

We have made changes in `**ldt- DAObs files and lis- dataassim/obs files** ` for SMAP data, should we also modify the code in any other files in ldt and lis apart from the ones mentioned here ? Or should we make changes in any other files in LISF module itself.
